src/isomorphic/classic/class/ReactClass.js

```diff
@@ -728,6 +728,11 @@ var ReactClassMixin = {
    * type signature and the only use case for this, is to avoid that.
    */
   replaceState: function(newState, callback) {
+    if (this.updater.isFiberUpdater) {
```

Comment:
Why does this strategy that Stack uses (two separate synchronous reconciliations) work for Fiber? My worry would be that this is covering up another bug.